### ISSUE TITLE
ValueError: function_call == None not supported (auto only)

### ISSUE THREAD

**Issue by philiprhoades** [cpacker/MemGPT#967]
**Describe the bug**
See log below.

Please describe your setup
 MemGPT version

What is the output of memgpt version?
0.3.0

How did you install memgpt?
pip install pymemgpt

What's your OS
Fedora Linux 38

How are you running memgpt?
tmux Terminal

I did note the Q that seemed to have cause the error this time (note: siblings names have been changed here):

```
> Enter your message: Very good! All my siblings are younger than me and, in order, they are: x, x, x, x, x and x.                                
An exception occurred when running agent.step():                                                                                                                              
Traceback (most recent call last):                                                                                                                                            
  File "/usr/local/lib/python3.11/site-packages/memgpt/agent.py", line 608, in step                                                                                           
    response = self._get_ai_reply(                                                                                                                                            
               ^^^^^^^^^^^^^^^^^^^                                                                                                                                            
  File "/usr/local/lib/python3.11/site-packages/memgpt/agent.py", line 362, in _get_ai_reply                                                                                  
    raise e                                                                                                                                                                   
  File "/usr/local/lib/python3.11/site-packages/memgpt/agent.py", line 342, in _get_ai_reply                                                                                  
    response = create(                                                                                                                                                        
               ^^^^^^^                                                                                                                                                        
  File "/usr/local/lib/python3.11/site-packages/memgpt/llm_api_tools.py", line 373, in wrapper                                                                                
    raise e                                                                                                                                                                   
  File "/usr/local/lib/python3.11/site-packages/memgpt/llm_api_tools.py", line 346, in wrapper                                                                                
    return func(*args, **kwargs)                                                                                                                                              
           ^^^^^^^^^^^^^^^^^^^^^                                                                                                                                              
  File "/usr/local/lib/python3.11/site-packages/memgpt/llm_api_tools.py", line 465, in create
    return get_chat_completion(
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memgpt/local_llm/chat_completion_proxy.py", line 159, in get_chat_completion
    result, usage = get_vllm_completion(endpoint, auth_type, auth_key, model, prompt, context_window, user)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memgpt/local_llm/vllm/api.py", line 17, in get_vllm_completion
    raise Exception(f"Request exceeds maximum context length ({prompt_tokens} > {context_window} tokens)")
Exception: Request exceeds maximum context length (16604 > 16384 tokens)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/memgpt/main.py", line 356, in run_agent_loop
    new_messages, user_message, skip_next_user_input = process_agent_step(user_message, no_verify)
                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memgpt/main.py", line 332, in process_agent_step
    new_messages, heartbeat_request, function_failed, token_warning, tokens_accumulated = memgpt_agent.step(
                                                                                          ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memgpt/agent.py", line 670, in step
    self.summarize_messages_inplace()
  File "/usr/local/lib/python3.11/site-packages/memgpt/agent.py", line 761, in summarize_messages_inplace
    summary = summarize_messages(agent_state=self.agent_state, message_sequence_to_summarize=message_sequence_to_summarize)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memgpt/memory.py", line 126, in summarize_messages
    response = create(
               ^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memgpt/llm_api_tools.py", line 373, in wrapper
    raise e
  File "/usr/local/lib/python3.11/site-packages/memgpt/llm_api_tools.py", line 346, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memgpt/llm_api_tools.py", line 465, in create
    return get_chat_completion(
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memgpt/local_llm/chat_completion_proxy.py", line 64, in get_chat_completion
    raise ValueError(f"function_call == {function_call} not supported (auto only)")
ValueError: function_call == None not supported (auto only)
? Retry agent.step()? (Y/n)
```

**Comment by starsnatched** [comment]
Your context length limit was reached. If you have a stack of Python traceback, make sure to look at the topmost one. You should try lowering the context length limit. You can achieve this via the CLI command `memgpt configure`. You can select the option 8,192 or 16,384 from the menu. It will warn the LLM to prepare for truncation once the current context reaches around 80% of the max context length.

**Comment by dlaliberte** [comment]
This sounds like it is closely related to what I reported, and fixed, in [#957.](s://github.com/cpacker/MemGPT/issues/957)

I avoided this problem with function_call being None just by commenting out the whole condition and error.  This allowed me to proceed to find the next problem re the memory limit being exceeded.

**Comment by philiprhoades** [comment]
> Your context length limit was reached. If you have a stack of Python traceback, make sure to look at the topmost one. You should try lowering the context length limit. You can achieve this via the CLI command `memgpt configure`. You can select the option 8,192 or 16,384 from the menu. It will warn the LLM to prepare for truncation once the current context reaches around 80% of the max context length.

Hmm . . I tried:

`memgpt run --agent phi_rho_05 --context-window 8192`
and
`memgpt run --agent phi_rho_05 --context-window 16384`

but I still got the same error but:

`memgpt run --agent phi_rho_05 --context-window 32768`

worked! - which seems to make sense to me - but I thought I read somewhere that we weren't supposed to use that value . .
